realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java

```diff
@@ -1019,4 +1022,167 @@ public void testRemoveAllFromRealmInvalidListShouldThrow() {
             assertEquals("This Realm instance has already been closed, making it unusable.", e.getMessage());
         }
     }
+
+    @Test
+    public void add_set_objectFromOtherThread() {
+        final CountDownLatch finishedLatch = new CountDownLatch(1);
+        final Dog dog = testRealm.where(Dog.class).findFirst();
+        final String expectedMsg = "Cannot copy an object from another Realm instance.";
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Realm realm = Realm.getInstance(testRealm.getConfiguration());
+                realm.beginTransaction();
+                RealmList<Dog> list = realm.createObject(Owner.class).getDogs();
+                list.add(realm.createObject(Dog.class));
+                try {
```

Comment:
@bmunkholm That was what i did before -- use common code https://github.com/realm/realm-java/pull/2390#discussion_r55023084 . But we found it is not easy to read.